.genai/io/capstone/rag/stdlib_docs.py

```
@@ -0,0 +1,459 @@
+        self._doc_embeddings = self.embeddings_client.embed_batch(self._doc_texts)
+
+        # Cache for next time
+        self._save_cache()
+
+    def get_doc_by_module(self, module_name: str) -> List[StdlibDoc]:
+        """Get all docs for a specific module."""
+        return [doc for doc in self.docs if doc.module == module_name]
+
+    def get_doc_by_name(self, name: str) -> Optional[StdlibDoc]:
+        """Get doc for a specific function/class name."""
+        for doc in self.docs:
+            if doc.name == name:
+                return doc
+        return None
+
+    def search_relevant_docs(
+        self,
+        query: str,
+        top_k: int = 3,
+        module_filter: List[str] = None
+    ) -> List[Tuple[StdlibDoc, float]]:
+        """
+        Search for relevant stdlib docs using semantic similarity.
+
+        Args:
+            query: Search query (code snippet or description)
+            top_k: Number of results to return
+            module_filter: If provided, only search these modules
+
+        Returns:
+            List of (StdlibDoc, score) tuples, sorted by relevance
+        """
+        if self._doc_embeddings is None:
+            self.initialize()
+
+        if self.embeddings_client is None:
+            # Fallback to keyword matching if no embeddings client
+            return self._keyword_search(query, top_k, module_filter)
+
+        # Get query embedding
+        query_embedding = self.embeddings_client.embed_text(query)
+
+        # Calculate cosine similarity with all docs
+        results = []
+        for i, (doc, doc_embedding) in enumerate(zip(self.docs, self._doc_embeddings)):
+            # Apply module filter
+            if module_filter and doc.module not in module_filter:
+                continue
+
+            # Cosine similarity
+            similarity = self._cosine_similarity(query_embedding, doc_embedding)
+            results.append((doc, similarity))
+
+        # Sort by similarity and return top_k
```

398 : The `except Exception:` clause is too broad. It catches and silences all errors during cache loading (e.g., file corruption, permission errors, malformed data), making it difficult to debug why the cache isn't being used. It's better to catch specific exceptions like `(IOError, pickle.PickleError, KeyError, EOFError)` and log the error.